src/de/grobox/liberario/fragments/DirectionsFragment.java

```diff
@@ -240,6 +244,23 @@ public void onResume() {
 		displayFavouriteTrips();
 		refreshAutocomplete(false);
 
+		// check if there's a preference for selected products for the current network
+		TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+		SharedPreferences sp = getContext().getSharedPreferences("pref_selected_products", 0);
+		String network_pref = current_network + "_selected_products";
+		if(sp.contains(network_pref)){
+			// get the selected products from the pref
+			Set<String> product_set = sp.getStringSet(network_pref, new HashSet<String>());
+            // TODO: convert Strings from product_set to Products
+            EnumSet<Product> h = EnumSet.noneOf(Product.class);
+            for(String s: product_set){
+                //h.add(??);
+            }
+			Serializable tmpProducts = h;
+            products = (EnumSet<Product>) tmpProducts;
+            onProductsChanged(products);
+		}
+
```

Comment:
I have problems finding a way for putting the products from the StringSet (SharedPreference) into the EnumSet<Product>. Can you tell me how to do this?
If this is solved, i'll clean up my code and update the pull request.